### DISCUSSION TITLE
How to calculate height from pressure using SCM output

### DISCUSSION THREAD

**Discussion by XiaSun-Atmos** [NCAR/ccpp-scm#537] [Q&A]
There are only pressure fields in SCM output. How to calculate altitudes at SCM model layers?

**Comment by hertneky** [comment]
If you are simply wanting to compute height for say...plotting purposes, you can use the formula from the function [get_height_from_pres](https://github.com/NCAR/ccpp-scm/blob/f944a103e807836e0ca9bae774744af6adca30f9/scm/etc/scripts/forcing_file_common.py#L90) that's in ccpp-scm/scm/etc/scripts/forcing_file_common.py. These functions were originally used by some of the forcing generator scripts.

**Comment by dustinswales** [comment]
@XiaSun-Atmos Two things come to mind. a) You could get the height from the hypsometric equation or b) follow what the SCM does internally ([compute Exner function, compute geopotential, then divide by gravity to get altitude](**https://github.com/NCAR/ccpp-scm/blob/main/scm/src/scm_vgrid.F90#L1615**))

**Comment by XiaSun-Atmos** [comment]
Thanks @hertneky @dustinswales .
I would like to keep consistent with what SCM does internally. Is it possible to point me to the file/module that includes this algorithm in SCM?

**Reply by dustinswales** [reply]
Sorry, see the [~25lines starting here](https://github.com/NCAR/ccpp-scm/blob/main/scm/src/scm_vgrid.F90#L1615).

**Reply by XiaSun-Atmos** [reply]
Thanks!
